scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -41,4 +53,176 @@ class CoverageRunner(
   ): List<String>? {
     return bazelClient.runCoverageForTestTarget(bazelTestTarget)
   }
+
+  private fun parseCoverageData(
+    coverageData: List<String>,
+    bazelTestTarget: String
+  ): CoverageReport {
+    var filePath = ""
+    var linesFound = 0
+    var linesHit = 0
+    val coveredLines = mutableListOf<CoveredLine>()
+    val branchCoverage = mutableListOf<BranchCoverage>()
+    val functionCoverage = mutableListOf<FunctionCoverage>()
+
+    var functionsFound = 0
+    var functionsHit = 0
+    var branchesFound = 0
+    var branchesHit = 0
+
+    var parseFile = false
```

Comment:
A cleaner option might be to do something like this:

```kotlin
val sfStartIdx = coverageData.indexOfFirst { it.startsWith("SF:") }
val eofIdx = coverageData.indexOfFirst { it.startsWith("end_of_record") }
val fileSpecificCovDatLines = coverageData.subList(sfStartIdx, eofIdx)
```

Now you can avoid the slightly awkward "am I parsing right now?" state of ``parseFile`` and eliminate a whole ``when`` block nesting below since we know exactly what lines to parse from ``fileSpecificCovDatLines``.

Reply:
Thanks Ben! Updated with the exact logic.